### DISCUSSION TITLE
Why does not PaginatorStream impl Stream?

### DISCUSSION THREAD

**Discussion by Ploppz** [awslabs/aws-sdk-rust#995] [Q&A]
Would be very convenient if it did. Because in code written with an earlier version of aws-sdk where it did impl Stream, I used the paginator from ListObjectV2 somewhere a `impl Stream` was required and I cannot do that anymore

**Comment by rcoh** [comment]
Good question! We removed this implementation because of the instability of the stream trait. We didn't want to commit to permanently exposing a pre-1.0 dependency that might eventually exist in `std`. This seems like a good candidate for addition to [`aws-smithy-types-convert`](https://docs.rs/aws-smithy-types-convert/0.60.0/aws_smithy_types_convert/) though!

**Reply by Ploppz** [reply]
Thanks!
I tried to implement it: https://github.com/smithy-lang/smithy-rs/pull/3299 Unsure about correctness and looking for feedback.
